### PR TITLE
JAMES-2862 Implement Remove address mapping in AddressMappingRoutes

### PR DESCRIPTION
**Pull request by billyfrost418**

### CODE REVIEW

File: server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/AddressMappingRoutesTest.java [hunk 1]
```diff
@@ -74,70 +77,135 @@ void stop() {
...
+        when()
+            .delete(MAPPING_SOURCE + "/targets/" + ALICE_ADDRESS)
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204);
+    }
+
+    @Test
+    void removeAddressMappingShouldReturnBadRequestWhenMappingSourceIsInvalid() {
+        when()
+            .delete( "random@domain@domain/targets/" + ALICE_ADDRESS)
```
Comment (Arsnael) [inline server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/AddressMappingRoutesTest.java]: Remove extra space

File: server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/AddressMappingRoutesTest.java [hunk 2]
```diff
@@ -74,70 +77,135 @@ void stop() {
...
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400);
+    }
+
+    @Test
+    void removeAddressMappingShouldReturnNotFoundWhenOneParameterIsEmpty() {
+        when()
+            .delete( MAPPING_SOURCE + "/targets/")
```
Comment (Arsnael) [inline server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/AddressMappingRoutesTest.java]: Idem

File: src/site/markdown/server/manage-webadmin.md [hunk 3]
```diff
@@ -1693,6 +1694,17 @@ Respond code:
...
+
+```
+curl -XDELETE http://ip:port/mappings/address/{mappingSource}/targets/{destinationAddress}
+```
+
+Remove an address mapping from the [Recipients rewrite table](/server/config-recipientrewritetable.html)
+Mapping source is the value of {mappingSource}
+Mapping destination is the value of {destinationAddress}
+Type of mapping destination is Address
+
```
Comment (Arsnael) [inline src/site/markdown/server/manage-webadmin.md]: Missing response codes

File: src/site/markdown/server/manage-webadmin.md [hunk 4]
```diff
@@ -1693,6 +1694,17 @@ Respond code:
 - 204: NO CONTENT
```
Comment (Arsnael) [inline src/site/markdown/server/manage-webadmin.md]: While we are at it,
s/Respond code/Response codes

**Comment by billyfrost418** [comment]
test this please

File: src/site/markdown/server/manage-webadmin.md [hunk 5]
```diff
@@ -1705,6 +1707,12 @@ Mapping source is the value of {mappingSource}
 Mapping destination is the value of {destinationAddress}
 Type of mapping destination is Address
 
+Response codes:
+
+- 204: NO CONTENT
+- 400: INVALID PARAMETERS
```
Comment (Arsnael) [inline src/site/markdown/server/manage-webadmin.md]: Why in upper case? (valid for other response codes too)

File: src/site/markdown/server/manage-webadmin.md [hunk 6]
```diff
@@ -1693,8 +1693,8 @@ Type of mapping destination is Address
 
 Response codes:
 
-- 204: NO CONTENT
-- 400: INVALID PARAMETERS
+- 204: No content
```
Comment (chibenwa) [inline src/site/markdown/server/manage-webadmin.md]: We should rather say "Action successfully performed"

File: src/site/markdown/server/manage-webadmin.md [hunk 7]
```diff
@@ -1709,8 +1709,8 @@ Type of mapping destination is Address
...
+- 400: Invalid parameters
 
 
 ## User Mappings
```
Comment (chibenwa) [inline src/site/markdown/server/manage-webadmin.md]: We should rather say "Action successfully performed"

**Comment by billyfrost418** [comment]
test this please

**Comment by billyfrost418** [comment]
test this please

**Comment by chibenwa** [comment]
(Just rebased)

**Comment by chibenwa** [comment]
`[ERROR]   JamesMailSpoolerTest.threadSuicideShouldAckTheItem:126 » ConditionTimeout Cond...`

test this please

**Comment by chibenwa** [comment]
```
[dfcabdde2aef12f09503dd2d2b57539e3d0d4dc3] [ERROR]   ICSAttachmentWorkflowTest.mailShouldNotContainCalendarContentInTextBodyButAttachment:801 
[dfcabdde2aef12f09503dd2d2b57539e3d0d4dc3] Expecting:
[dfcabdde2aef12f09503dd2d2b57539e3d0d4dc3]  <"AAAD OK FETCH completed.
[dfcabdde2aef12f09503dd2d2b57539e3d0d4dc3] ">
[dfcabdde2aef12f09503dd2d2b57539e3d0d4dc3] to contain:
[dfcabdde2aef12f09503dd2d2b57539e3d0d4dc3]  <["Content-Type: multipart/mixed", "Content-Disposition: attachment"]>
[dfcabdde2aef12f09503dd2d2b57539e3d0d4dc3] but could not find:
[dfcabdde2aef12f09503dd2d2b57539e3d0d4dc3]  <["Content-Type: multipart/mixed", "Content-Disposition: attachment"]>
```

Not related

test this please

**Comment by chibenwa** [comment]
`ConcurrentTestRunnerTest.runShouldPerformAllOperationsEvenOnOccasionalExceptions:230 » NotTerminated`

test this please

**Comment by Arsnael** [comment]
```
[ERROR]   ConcurrentTestRunnerTest.runShouldPerformAllOperationsEvenOnOccasionalExceptions:230 » NotTerminated
```
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**